doc/source/whatsnew/v0.21.0.txt

```diff
@@ -565,6 +565,7 @@ Indexing
 - Bug in intersection of ``RangeIndex`` with negative step (:issue:`17296`)
 - Bug in ``IntervalIndex`` where performing a scalar lookup fails for included right endpoints of non-overlapping monotonic decreasing indexes (:issue:`16417`, :issue:`17271`)
 - Bug in :meth:`DataFrame.first_valid_index` and :meth:`DataFrame.last_valid_index` when no valid entry (:issue:`17400`)
+- Bug in ``Series.rename`` when called with a `callable` alters name of series rather than index of series. (:issue:`17407`)
```

Comment:
Bug in `` :func:`Series.rename` `` when called with a callable, incorrectly alters the name of the ``Series``, rather than the name of the ``Index``